calour/io.py

```diff
@@ -78,6 +132,59 @@ def _read_table(f):
     return table
 
 
+def read_open_ms(data_file, sample_metadata_file=None, feature_metadata_file=None,
+                 normalize=False, rescale=10000, description=None, sparse=False, **kwargs):
+    '''Load an OpenMS metabolomics experiment
+
+    Parameters
+    ----------
+    data_file : str
+        name of the OpenMS bucket table CSV file
+    sample_metadata_file : str or None (optional)
+        None (default) to not load metadata per sample
+        str to specify name of sample mapping file (tsv)
+    feature_metadata_file : str or None (optional)
+        Name of table containing additional metadata about each feature
+        None (default) to not load
+    normalize : bool (optional)
+        True to normalize each sample to constant sum,
+        False (default) to not normalize
+    rescale : int or None (optional)
+        int to rescale reads so mean total reads per sample (over
+        all samples) to value rescale,
+        None to not rescale
+    description : str or None (optional)
+        Name of the experiment (for display purposes).
+        None (default) to assign file name
+    sparse : bool (optional)
+        False (default) to store data as dense matrix (faster but more memory)
+        True to store as sparse (CSR)
+
+    Returns
+    -------
+    exp : ``Experiment``
+    '''
+    logger.info('Reading OpenMS data (OpenMS bucket table %s, map file %s)' % (data_file, sample_metadata_file))
+    exp = read(data_file, sample_metadata_file, feature_metadata_file,
+               data_file_type='openms', sparse=sparse, **kwargs)
+    # record the original total read count into sample metadata
+    if normalize:
+        exp.normalize(inplace=True, total=rescale)
+    elif rescale:
+        exp.normalize(inplace=True, total=rescale)
+
+    exp.sample_metadata['id'] = exp.sample_metadata.index.values
```

Comment:
useful to have id as a column besides the index (also for feature_metadata), since then we can choose to display it in the plot() function with qt/jupyter ui